src/textfile_gui.cpp

```diff
@@ -344,48 +339,48 @@ static void Xunzip(byte **bufp, size_t *sizep)
 	FILE *handle = FioFOpenFile(textfile, "rb", dir, &filesize);
 	if (handle == nullptr) return;
 
-	this->text = ReallocT(this->text, filesize);
-	size_t read = fread(this->text, 1, filesize, handle);
+	char *buf = MallocT<char>(filesize);
+	size_t read = fread(buf, 1, filesize, handle);
 	fclose(handle);
 
 	if (read != filesize) return;
 
 #if defined(WITH_ZLIB)
 	/* In-place gunzip */
-	if (StrEndsWith(textfile, ".gz")) Gunzip((byte**)&this->text, &filesize);
+	if (StrEndsWith(textfile, ".gz")) Gunzip((byte**)&buf, &filesize);
 #endif
 
 #if defined(WITH_LIBLZMA)
 	/* In-place xunzip */
-	if (StrEndsWith(textfile, ".xz")) Xunzip((byte**)&this->text, &filesize);
+	if (StrEndsWith(textfile, ".xz")) Xunzip((byte**)&buf, &filesize);
 #endif
 
-	if (!this->text) return;
+	if (buf == nullptr) return;
 
-	/* Add space for trailing \0 */
-	this->text = ReallocT(this->text, filesize + 1);
-	this->text[filesize] = '\0';
+	this->text = std::string(buf, filesize);
+	free(buf);
 
 	/* Replace tabs and line feeds with a space since StrMakeValidInPlace removes those. */
-	for (char *p = this->text; *p != '\0'; p++) {
-		if (*p == '\t' || *p == '\r') *p = ' ';
+	for (auto &c : this->text) {
+		if (c == '\t' || c == '\r') c = ' ';
 	}
 
 	/* Check for the byte-order-mark, and skip it if needed. */
-	char *p = this->text + (strncmp(u8"\ufeff", this->text, 3) == 0 ? 3 : 0);
-
-	/* Make sure the string is a valid UTF-8 sequence. */
-	StrMakeValidInPlace(p, this->text + filesize, SVS_REPLACE_WITH_QUESTION_MARK | SVS_ALLOW_NEWLINE);
+	if (StrStartsWith(this->text, u8"\ufeff")) this->text = this->text.erase(0, 3);
+	this->text = StrMakeValid(this->text, SVS_REPLACE_WITH_QUESTION_MARK | SVS_ALLOW_NEWLINE);
```

Comment:
As mentioned on IRC, this is not as easy as you make it sound :) The replacing of `\t` and `\r` is a bit of an issue. You can't do that on a `string_view` obviously, and it needs to be done before `StrMakeValid`. So as a compromise I can remove at least one copy, by doing the BOM earlier, as you suggested :) But I do not see how to avoid 2 copies, without doing the `\t` and `\r` replacement in the C-world again (which I kinda want to avoid :D).

Ideally we had a `StrMakeValidInPlace` that works on `std::string`, but we don't (yet?).

Open to other suggestions, ofc :)